src/main/java/org/truffleruby/parser/TranslatorEnvironment.java

```diff
@@ -161,11 +160,7 @@ public ReadLocalNode findFrameLocalGlobalVarNode(String name, Source source, Sou
         final FrameSlot slot = methodScope.declareVar(name);
 
         final LocalVariableType type;
-        if (GlobalVariables.ALWAYS_DEFINED_GLOBALS.contains(name)) {
-            type = LocalVariableType.ALWAYS_DEFINED_GLOBAL;
-        } else {
-            type = LocalVariableType.FRAME_LOCAL_GLOBAL;
-        }
+        type = LocalVariableType.FRAME_LOCAL_GLOBAL;
```

Comment:
Can be one line with the declaration of the local var, or even inlined below.
`GlobalVariables.ALWAYS_DEFINED_GLOBALS` can also be removed I'd guess (can be another PR).